userspace/libsinsp/k8s.cpp

```diff
@@ -38,12 +38,16 @@ k8s::k8s(const std::string& uri, bool is_captured,
 #endif // HAS_CAPTURE
 		filter_ptr_t event_filter,
 		ext_list_ptr_t extensions,
-		bool events_only) :
+		bool events_only
+#ifdef HAS_CAPTURE
+		,const std::string& node_selector
+#endif // HAS_CAPTURE
```

Comment:
When is the k8s code relevant for !HAS_CAPTURE? I know these are not the first `#ifdef HAS_CAPTURE`s here but do we ever talk to a k8s server outside a live capture?

Reply:
I couldn't remove the ifdef only from here since the code point where the filtering takes place is already within a `HAS_CAPTURE` block :point_down: 
https://github.com/falcosecurity/libs/pull/49/files#diff-b2668de87a1d8591041c8be578efe99db5eeaab8d2394db3d6a36f3881870bcbR93-R96

Generally speaking, I agree with you it should be simplified if we never talk to k8s outside a live capture, but I'm not sure about that :smile_cat: 
However, I believe this kind of improvement should be handled in another PR eventually. Wdyt?

Reply:
Yes, removing the ifdef will probably bubble up to quite a few places. A separate PR is okay if the ifdef patch isn't trivial